NCPCore/src/main/java/fr/neatmonster/nocheatplus/checks/blockplace/Against.java

```diff
@@ -70,17 +75,20 @@ else if (!pData.hasPermission(Permissions.BLOCKPLACE_AGAINST_AIR, player)) {
         else if (BlockProperties.isLiquid(againstType)) {
             // TODO: F_PLACE_AGAINST_WATER|LIQUID...
             if ((placedMat != BridgeMaterial.LILY_PAD
-                    || !BlockProperties.isLiquid(block.getRelative(BlockFace.DOWN).getType()))
-                    && !BlockProperties.isNewLiq(bdata.getLastType())
-                    && !pData.hasPermission(Permissions.BLOCKPLACE_AGAINST_LIQUIDS, player)) {
+                 || !BlockProperties.isLiquid(block.getRelative(BlockFace.DOWN).getType()))
+                 && !BlockProperties.isNewLiq(bdata.getLastType())
+                 && !pData.hasPermission(Permissions.BLOCKPLACE_AGAINST_LIQUIDS, player)) {
                 violation = true;
             }
-        } else
+        }
         // Replace block placed by block placed and interact with air or water 
-        if (block.equals(blockAgainst) && (bdata.getLastType() == null || (BlockProperties.isLiquid(bdata.getLastType()) && !BlockProperties.isNewLiq(bdata.getLastType())))
-        	&& !pData.hasPermission(Permissions.BLOCKPLACE_AGAINST_SELF, player)) {
+        else if (block.equals(blockAgainst) && (bdata.getLastType() == null || (BlockProperties.isLiquid(bdata.getLastType())
+                && !BlockProperties.isNewLiq(bdata.getLastType())))
+        	    && !pData.hasPermission(Permissions.BLOCKPLACE_AGAINST_SELF, player) 
+                && placedMat != BridgeMaterial.LILY_PAD) {

```

Comment:
Checking if the block is a lily pad here is not needed as it's already checked above at line 77

Reply:
Aren't those  separate else if statements? The "Self" submodule also needs to be skipped if the placed block is a lily pad, it fixes [#77](https://github.com/Updated-NoCheatPlus/NoCheatPlus/issues/77). 

```
   else if (BlockProperties.isLiquid(againstType)){
        if ((placedMat != BridgeMaterial.LILY_PAD && (...)){
            (...)
        }
   }
   else if (block.equals(blockAgainst) && (...) && placedMat != BridgeMaterial.LILY_PAD){
            (...)
   }
```

Reply:
Yeah, you're right. Although I'm slightly confused as to what this is preventing as the comment left is worded weirdly. What exactly is the "self" check doing? 

Judging from the bug report it was preventing users from placing lilypads on water? If so then something is broken above that needs to be fixed, since it shouldn't reach this statement unless isLiquid(againstType) is returning false for water.

Reply:
So, apparently using event#getBlockAgainst() returns the block that has just been placed before returning the liquid type, so BlockProperties#isLiquid(againstType) will always return false, using matAgainst however will return the correct block type.

https://user-images.githubusercontent.com/20213873/106717307-d506f180-65ff-11eb-9b20-e315085782c1.mov